### PR TITLE
render 404 and give flash alert for expired reset token

### PR DESCRIPTION
**Pull request by BintLopez**
## Changelog
- Adds error handling for stale reset token

## Link to issue:  
Fixes issue https://github.com/MidwestAccessCoalition/lockbox_rails/issues/357

## Steps for QA/Special Notes:
- Adding automated tests

## Relevant Screenshots: 
image

## Are you ready for review?:

- [x] Added relevant tests
- [x] Linked PR to the issue
- [ ] Added notes for QA/special notes
- [ ] Added revelant screenshots

### CODE REVIEW

**Comment by misternu** [comment]
@BintLopez should be gtg

File: app/controllers/users/passwords_controller.rb [hunk 1]
```diff
@@ -69,9 +68,19 @@ def set_existing_user
...
-    token = Devise.token_generator.digest(
-      self, :reset_password_token, params[:reset_password_token]
-    )
-    @existing_user = User.find_by(reset_password_token: token)
+    @existing_user = User.with_reset_password_token(reset_password_token)
+    record_not_found unless @existing_user && @existing_user.reset_password_period_valid?
+  end
+
+  def reset_password_token
+    params[:reset_password_token] || params[:user][:reset_password_token]
```
Comment (BintLopez) [inline app/controllers/users/passwords_controller.rb]: When do we expect the reset token to be under `params` vs `params[:user]`?

Comment (misternu) [reply app/controllers/users/passwords_controller.rb]: when the user clicks the link in their email eg
`<a href="http://localhost:3000/users/password/edit?reset_password_token=ye5smKv3z46yYyzCMekz">Change my password</a>`

Comment (misternu) [reply app/controllers/users/passwords_controller.rb]: but in the edit form it's nested, and the devise controller seems to expect it these two ways

**Pull request merged**